core/server/services/auth/api-key/admin.js

```diff
@@ -0,0 +1,82 @@
+const jwt = require('jsonwebtoken');
+const models = require('../../../models');
+const {
+    InternalServerError,
+    UnauthorizedError
+} = require('../../../lib/common/errors');
+
+const JWT_OPTIONS = {
+    maxAge: '5m',
+    algorithms: ['HS256']
+};
+
+/**
+ * Remove 'Bearer' from raw authorization header and extract the API Key ID and
+ * JWT token. Eg. Authorization: Bearer ${objectId}:${JWT}
+ * @param {string} header
+ */
+const extractCredentialsFromHeader = function extractCredentialsFromHeader(header) {
+    let [scheme, credentials] = header.split(' ');
+
+    if (/^Bearer$/i.test(scheme)) {
+        let [apiKeyId, token] = credentials.split('|');
+
+        return {
+            apiKeyId,
+            token
+        };
+    } else {
+        return {};
+    }
+};
+
+const authenticateAdminAPIKey = function authenticateAdminAPIKey(req, res, next) {
+    // allow fallthrough to other auth methods or final ensureAuthenticated check
+    if (!req.headers || (req.headers && !req.headers.authorization)) {
+        return next();
+    }
+
+    let {apiKeyId, token} = extractCredentialsFromHeader(req.headers.authorization);
```

Comment:
This can be a `const`

Reply:
Do we have a standard for that on the server-side code? In Ghost-Admin this would always be `let` as the code there subscribes to the reasoning laid out in https://madhatted.com/2016/1/25/let-it-be

Reply:
We seem to have a `liberal let` according to - https://github.com/TryGhost/Ghost/blob/master/.eslintrc.json#L243
But personally, prefer doing `constantly const` as it's easier enforced by eslint, so there's no room fot sometimes using `let` and sometimes `const` :smiley:  
Probably @kirrg001 has more insight into the rule here

Reply:
> always be let as the code there subscribes to the reasoning laid out in https://madhatted.com/2016/1/25/let-it-be

I am not 100% convinced of this article. Just my personal opinion :)

---

We are currently following **constantly const** in the backend.

The [official node.js docs/guides](https://nodejs.org/en/docs/guides/anatomy-of-an-http-transaction/) use constantly const too. [This medium post](https://medium.com/@bkinsey808/why-i-prefer-constantly-const-over-liberal-let-476d228f8df9) collected some links for constantly const vs liberal let.